### DISCUSSION TITLE
Error in getStaticProps

### DISCUSSION THREAD

**Discussion by Ali0011k** [vercel/next.js#68320] [Help] [answered]
### Summary

Hi everyone, I'm encountering an issue with my Next.js project. When I'm using getStaticProps to fetch data, I'm running into the following error:
`Error: getStaticProps is missing the following required keys: revalidate`

### Additional information

```js
Here is my code:
`export async function getStaticProps() {
  const res = await fetch('https://api.example.com/data');
  const data = await res.json();

  return {
    props: {
      data,
    },
  };
}

function HomePage({ data }) {
  return (
    <div>
      <h1>Data from API</h1>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </div>
  );
}

export default HomePage;
`

Could someone please help me figure out how to resolve this issue? Thanks in advance!
```

**Comment by Mr-Meshky** [comment, marked as answer]
Hi,

The error you're seeing suggests that you're missing the revalidate key in your getStaticProps return object. The revalidate key is used to specify the revalidation time for Incremental Static Regeneration (ISR). This allows your static pages to be regenerated in the background at a specified interval.

Here's how you can modify your getStaticProps to include the revalidate key:

```
export async function getStaticProps() {
  const res = await fetch('https://api.example.com/data');
  const data = await res.json();

  return {
    props: {
      data,
    },
    revalidate: 10, // Revalidate the page every 10 seconds
  };
}

function HomePage({ data }) {
  return (
    <div>
      <h1>Data from API</h1>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </div>
  );
}

export default HomePage;
```

By adding revalidate: 10, you ensure that your page will be regenerated at most once every 10 seconds if there are requests for it.

If you do not need ISR and your data doesn't change often, you can set revalidate to a larger value, or even skip it if you're fine with the static generation happening only at build time.

Let me know if this helps!

**Comment by icyJoseph** [comment]
What version of Nextjs are you using? The revalidate prop is not mandatory AFAIK, so it is weird that it fails your build, because of it.
